src/components/Stepper.tsx

```diff
@@ -22,13 +22,8 @@ export function Stepper({ steps, currentStep, onChange }: StepperBarProps) {
     throw new Error("Stepper must be initialized with at least one step");
   }
 
-  const [progressBarStep, setProgressBarStep] = useState(0);
-  useEffect(() => {
-    const currentStepIndex = steps.findIndex((s: Step) => s.value === currentStep) || 0;
-    if (currentStepIndex > progressBarStep) {
-      setProgressBarStep(currentStepIndex);
-    }
-  }, [currentStep]);
+  // calc progress based on last completed step - return -1 when no steps completed
+  const progressBarStep = steps.reduce((acc, step, idx) => (step.state === "complete" ? idx : acc), -1);
```

Comment:
If I understand this right, then the progress bar will show no progress when the user is on the first step. The progress bar is no longer reflecting what step the user is on, but which steps the user has completed, correct? This is a pretty big difference from the original specs for this component. I'm okay with it, just want to make sure that is intended.